src/solver/DiffusionSolver.cpp

```diff
@@ -6,20 +6,20 @@
 
 #include "DiffusionSolver.h"
 #include "../interfaces/IDiffusion.h"
-#include "../utility/Parameters.h"
 #include "../Domain.h"
 #include "SolverSelection.h"
 
-DiffusionSolver::DiffusionSolver(FieldController *field_controller) {
+DiffusionSolver::DiffusionSolver(Settings const &settings, FieldController *field_controller) :
+        m_settings(settings) {
 #ifndef BENCHMARKING
-    m_logger = Utility::create_logger(typeid(this).name());
+    m_logger = Utility::create_logger(m_settings, typeid(this).name());
 #endif
     m_field_controller = field_controller;
-    auto params = Parameters::getInstance();
-    std::string diffusionType = params->get("solver/diffusion/type");
-    SolverSelection::SetDiffusionSolver(&this->dif, diffusionType);
 
-    m_nu = params->get_real("physical_parameters/nu");
+    std::string diffusionType = m_settings.get("solver/diffusion/type");
+    SolverSelection::SetDiffusionSolver(m_settings, &this->dif, diffusionType);
+
+    m_nu = m_settings.get_real("physical_parameters/nu");
```

Comment:
remove member variable `m_nu` instead access value through Settings

Reply:
done in 0d1d248